### PR TITLE
Replace Jodatime with Java time and Scalaz with cats

### PR DESCRIPTION
**Pull request by kiequoo**
This leaves a cleaner API that only deals in classes from the standard library.

Also, updated all the dependencies and added some more type safety.

### CODE REVIEW

File: msgs/src/main/scala/com/thenewmotion/ocpi/OcpiDateTimeParser.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.time.{ZoneOffset, ZonedDateTime}
+import java.time.format.{DateTimeFormatter, DateTimeFormatterBuilder}
+import DateTimeFormatter._
+
+import scala.util.Try
+
+object OcpiDateTimeParser {
+
+  private val formatter =
```
Comment (DStranger) [inline msgs/src/main/scala/com/thenewmotion/ocpi/OcpiDateTimeParser.scala]: Can't this be replaced with `DateTimeFormatter.ISO_INSTANT`?

File: build.sbt [hunk 2]
```diff
@@ -19,7 +19,7 @@ val akka =
     akkaModule("http-spray-json")
   )
 
-val scalaz = Seq("org.scalaz"        %% "scalaz-core"         %   "7.2.14")
+val cats = Seq("org.typelevel" %% "cats" % "0.9.0")
```
Comment (DStranger) [inline build.sbt]: There's a `1.0.0-MF` already, we should probably migrate.

Comment (DStranger) [inline build.sbt]: It breaks a lot of things though.

File: msgs/src/main/scala/com/thenewmotion/ocpi/OcpiDateTimeParser.scala (hunk 1 above)
Comment (kiequoo) [inline msgs/src/main/scala/com/thenewmotion/ocpi/OcpiDateTimeParser.scala]: The trailing Z is optional according the OCPI spec, but not with this formatter

**Pull request merged**